### PR TITLE
Optimized the calculation of ReadOnly paths from the model Plugins Fixes #525

### LINKED ISSUE #525: Improve performance of gathering readonly paths from Model Plugins
**Is your feature request related to a problem? Please describe.**
When extracting "readonly paths" from Model Plugins, the system analyses every leaf in the system. Instead it should stop when it finds a container or list marked "config false" because there can only be "config false" items below that. Currently for the devicesim model there are **175** paths - this should be something like 50

**Describe the solution you'd like**
Do the task more efficiently.

**Describe alternatives you've considered**
Not doing it - leaving it as it is will an impact on future features that use this list

**Additional context**
The list of read only paths will be used to set up a subscribe to devices - for this reason we would like the list of paths to be as concise as possible

### PR DESCRIPTION
**Pull request by SeanCondon**

### CODE REVIEW

File: docs/modelplugin.md [hunk 1]
```diff
@@ -215,3 +209,35 @@ was needed.
 > If the generator program reports that a dependency was required e.g.
 **openconfig-inet-types.yang** then the version of this file with a date equal
 to or before 2017-07-14 should be downloaded - it is **openconfig-inet-types@2017-07-14.yang**
+
+## Readonly paths in OpenConfig models
```
Comment (Andrea-Campanella) [inline docs/modelplugin.md]: I think this is valid for all models not just openconfig.

File: pkg/northbound/admin/admin.go [hunk 2]
```diff
@@ -76,13 +78,23 @@ func (s Server) ListRegisteredModels(req *proto.ListModelsRequest, stream proto.
 			schemaEntries = append(schemaEntries, &schemaEntry)
 		}
 
+		roPaths := make([]string, 0)
+		if req.Verbose {
+			var ok bool
```
Comment (Andrea-Campanella) [inline pkg/northbound/admin/admin.go]: why do you need to define the ok variable ?

File: docs/modelplugin.md (hunk 1 above)
Comment (SeanCondon) [reply docs/modelplugin.md]: This is specifically an OpenConfig section of the document, so I thought it was appropriate

File: pkg/northbound/admin/admin.go (hunk 2 above)
Comment (SeanCondon) [reply pkg/northbound/admin/admin.go]: because I can't use ":=" in the next line because roPaths exists already

Comment (SeanCondon) [reply pkg/northbound/admin/admin.go]: because I can't use ":=" in the next line

Comment (Andrea-Campanella) [reply pkg/northbound/admin/admin.go]: you can still do := if there is at least one new variable to the left of the assignment.

Comment (SeanCondon) [reply pkg/northbound/admin/admin.go]: there are bigger issues than this

Comment (SeanCondon) [reply pkg/northbound/admin/admin.go]: doesn't work in this case

**Pull request merged**